src/Dispatcher/DeferredDispatcher.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Yiisoft\EventDispatcher\Dispatcher;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+
+final class DeferredDispatcher implements EventDispatcherInterface
+{
+    private EventDispatcherInterface $dispatcher;
+    private array $events = [];
+
+    public function __construct(EventDispatcherInterface $dispatcher)
+    {
+        $this->dispatcher = $dispatcher;
+    }
+
+
+    public function dispatch(object $event)
+    {
+        $this->events[] = $event;
```

Comment:
`dispath` should return event by [interface](https://github.com/php-fig/event-dispatcher/blob/3ef040d7c05652f55bd05115baf059a445cfc79c/src/EventDispatcherInterface.php#L17).